pipeline/ffmpeg.go

```diff
@@ -34,6 +37,24 @@ func (f *ffmpeg) HandleStartUploadJob(job *JobInfo) (*HandlerOutput, error) {
 	job.SourceOutputURL = sourceOutputURL.String()
 	job.SegmentingTargetURL = segmentingTargetURL.String()
 	log.AddContext(job.RequestID, "segmented_url", job.SegmentingTargetURL)
+	job.ReportProgress(clients.TranscodeStatusPreparing, 0.3)
+
+	// Create a temporary local file to write to
+	localSourceFile, err := os.CreateTemp(os.TempDir(), "source*")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create local file (%s) for segmenting: %s", localSourceFile.Name(), err)
+	}
+	defer localSourceFile.Close()
+	defer os.Remove(localSourceFile.Name()) // Clean up the file as soon as we're done segmenting
+
+	// Copy the file locally because of issues with ffmpeg segmenting and remote files
+	// We can be aggressive with the timeout because we're copying from cloud storage
+	timeout, cancel := context.WithTimeout(context.Background(), 30*time.Minute)
```

Comment:
I think the use of `context.Background()` here is going to leak a file if we shut down the server in the middle of a download — would be better if we got passed in a parent context that will properly shut down on server close

Reply:
Discussed offline - I'm going to do a followup PR to have Catalyst API clean up files that match this pattern on startup